### PR TITLE
 add support for jpeg optimal huffman encoding

### PR DESCRIPTION
**Pull request by pineapplespatula**
Turns flite and jpeg optimal encoding on by default as well 
requires https://github.com/adobe-photoshop/generator-core/pull/334

### CODE REVIEW

**Comment by mcilroyc** [comment]
1. Would it be more flexible to allow a named jpeg encoding scheme (with "optimal" being the default) instead of just a bool switch for "optimal"?
1. We can/should bump the generator minor version there, and bump the dependency here in package.json
1. I'm not sure I want to bundle the defaulting of flite to true in this PR ... can you argue that it is necessary?
1. This should still work with (or at least gracefully ignore) the jpeg encoding parameter even if using ImageMagic.  (I think image magic has a similar-but-not-the-same structure for supplying this parameter) (I guess this is really a comment for the generator-core change).

**Comment by pineapplespatula** [comment]
1. yes if maximum fexibility is what we're going for, but optimal is the only option surfaced by save for web
2. in DS we seem to do that after the PR has gone in, but i can make it a part of this one if you think that would be preferable
3. It isn't necessary to do in this PR, but i am in favor of defaulting it to true in general
4. I cannot find an encoding option for jpg in imagemagik, and this should ingore the option in the case where flite isn't enabled (in core)

**Comment by mcilroyc** [comment]
1. Please make it flexible
1. My suggestion: After the gen-core PR is merged, I'll bump the version.  THEN in the PR bump the dependency listed in package.json before we merge this PR
1. Please leave it off for now, and we'll flip the bit when that Story comes up in JIRA (meaning: when QE is prepared for the load)
1. ignoring it is fine (I see now that it will ignore it in gen-core)

**Comment by pineapplespatula** [comment]
back to @mcilroyc

File: lib/renderer.js [hunk 1]
```diff
@@ -117,6 +121,12 @@
     }
 
     /**
+     * @type {String=}
+     * String witht he jpeg encoding string to use
```
Comment (mcilroyc) [inline lib/renderer.js]: typo :steam_locomotive:

**Pull request merged**